mqtt/mqtt-bridge/src/persist/waking_state/waking_store.rs

```diff
@@ -0,0 +1,109 @@
+#![allow(dead_code)] // TODO remove when ready
+
+use std::{collections::hash_map::Entry, collections::HashMap, task::Waker};
+
+use bincode::{self};
+use mqtt3::proto::Publication;
+use rocksdb::{IteratorMode, Options, DB};
+use uuid::Uuid;
+
+use crate::persist::{waking_state::StreamWakeableState, Key, PersistError};
+
+/// When elements are retrieved they are added to the in flight collection, but kept in the original db store.
+/// Only when elements are removed from the in-flight collection they will be removed from the store.
+pub struct WakingStore {
+    db: DB,
+    in_flight: HashMap<Key, Publication>,
+    waker: Option<Waker>,
+    column_family: String,
```

Comment:
I tried to store the ColumnFamily struct but ran into issues. You get it back as a reference from the rocksdb api and there are difficulties moving it around. Adding lifetimes isn't enough as WakingStore needs to be Send/Sync so we would need to wrap in a mutex.

I decided it was better to store the information necessary to access the column family, rather than the &ColumnFamily itself.